azureenergylabelerlib/validations.py

```diff
@@ -53,6 +55,60 @@
 LOGGER.addHandler(logging.NullHandler())
 
 
+def is_valid_resource_group_name(resource_group_name):
+    """Checks whether a provided resource group name is a valid resource group name.
+
+    Args:
+        resource group name (str): A resource group name string.
+
+    Returns:
+        True if the provided value is a valid Azure resource group name, false otherwise.
+
+    """
+    pattern = r'^(?!.*\.$)[a-zA-Z0-9._()-]{1,90}$'
+    if not re.match(pattern, resource_group_name):
+        return False
+    return True
+
+
+def are_valid_resource_group_names(resource_group_name):
+    """Checks whether the provided list contains only valid Azure resource group names.
+
+    Args:
+        resource_group_name (list): A list of resource group names strings.
+
+    Returns:
+        True if the provided list contains only valid Azure resource group name, false otherwise.
+
+    """
+    return all(is_valid_resource_group_name(resource_group_name) for resource_group_name in resource_group_name)
```

Comment:
This is buggy, since we are iterating over the same resource, this only getting each character of the input string.